api/envoy/api/v2/listener/listener.proto

```diff
@@ -88,6 +89,28 @@ message FilterChainMatch {
   // [#not-implemented-hide:]
   google.protobuf.UInt32Value suffix_len = 5;
 
+  enum RequestSourceType {
+    // Any request origin matches
+    ANY = 0;
+    // Match a request that originates from the same host network or namespace
```

Comment:
Match a request with same source and destination IP. In simple network configurations, this is likely to be a same host origination... (and the rest of what you have today).